packages/vite/src/node/plugin.ts

```diff
@@ -179,3 +179,7 @@ export interface Plugin<A = any> extends RollupPlugin<A> {
 }
 
 export type HookHandler<T> = T extends ObjectHook<infer H> ? H : T
+
+export type WithRequiredHook<K extends keyof Plugin> = Plugin & {
```

Comment:
Maybe we could call this one `PluginWithRequiredHook`

Reply:
Sure, longer name but with more readability. I've renamed it.